enumerable_method.rb

```diff
@@ -0,0 +1,124 @@
+public def my_each(arr)
+    for name in arr do
+        yield (name)
+    end
+end
+
+names = ["Amy", "John", "Joe"]
+my_each(names) {|a| puts "Hello #{a}"}
+
+
+names = %w[Amy John Joe]
+
+my_each(names) { |a| puts "Hello #{a}" }
+
+def my_each_with_index(arr)
+  len = arr.length
+
+  i = 1
+
+  my_each(arr) do |item|
+    yield item if i <= len && i.even?
+
+    i += 1
+  end
+end
+
+grocery_list = %w[banana plum chocolate toiletries]
+
+my_each_with_index(grocery_list) { |m| puts "Please get #{m}" }
+
+def my_select(arr)
+  new_array = []
+    for val in arr do  
+      unless val.is_a? Integer
+        new_array.push(val)
+      end
+    end
+    p new_array
+end
```

Comment:
#my_select returns an array containing all elements of enum for which the given block returns 
a true value
#my_select returns an array containing all elements of enum for which the given block returns a true value::range
#my_select returns an enumerator if no block is given
#my_select does not mutate the original array